akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -411,8 +427,36 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
 
   private def closeScheduler(): Unit = scheduler match {
     case x: Closeable ⇒ x.close()
-    case _            ⇒
+    case _            ⇒ // ignore, this is fine
+  }
+
+  // ========================================================
+  // ================== OLDEST MAINTANANCE ==================
+  // ========================================================
+
+  /** INTERNAL API: Returns state.members sorted by Member.ageOrdering (from oldest to youngest) */
+  @InternalApi private[cluster] def membersByAge(): immutable.SortedSet[Member] =
+    SortedSet.empty(Member.ageOrdering) ++ state.members.toList // FIXME
+
+  /** INTERNAL API */
+  @InternalApi
+  private[cluster] def isAtLeastNthOldestAddress(address: UniqueAddress, memberFilter: Member ⇒ Boolean, n: Int): Boolean = {
+    require(n > 0, "n must be > 0 (oldest is 1, 2nd oldest is 2, ...)")
+
+    val sortedMembers = membersByAge()
+    val foundPos = sortedMembers.filter(memberFilter).zipWithIndex.find(_._1.uniqueAddress == UniqueAddress).map(_._2).getOrElse(-1)
+    foundPos <= n
+  }
+  /** INTERNAL API */
+  @InternalApi
+  private[cluster] def isAtLeastNthOldestMember(member: Member, memberFilter: Member ⇒ Boolean, n: Int): Boolean = {
+    require(n > 0, "n must be > 0 (oldest is 1, 2nd oldest is 2, ...)")
+    isAtLeastNthOldestAddress(member.uniqueAddress, memberFilter, n)
   }
+  /** INTERNAL API */
+  @InternalApi
+  private[cluster] def takeNOldestMembers(memberFilter: Member ⇒ Boolean, n: Int): immutable.SortedSet[Member] =
+    membersByAge().filter(memberFilter).take(n)
```

Comment:
not sure where the best place to put those is, but would be nice to have them in a common place I think